overlay.d/05core/usr/lib/dracut/modules.d/35coreos-ignition/module-setup.sh

```diff
@@ -23,9 +23,19 @@ install() {
         lsblk \
         sed \
         grep \
-        sgdisk \
         uname
 
+
+    # In RHEL10 / C10s sgdisk got dropped so we get sgdisk from ignition
+    # as it's vendored there.
+    # See https://issues.redhat.com/browse/RHEL-56080
+    source /usr/lib/os-release
+    if [ "$PLATFORM_ID" = "platform:el10" ]; then
+        inst /usr/libexec/ignition-sgdisk /usr/sbin/sgdisk
+    else
+        inst sgdisk
+    fi
```

Comment:
Maybe cleaner here would be to not do a version check here but rather

```suggestion
    # In some cases we had to vendor gdisk in Ignition. If we detect
    # that case here then use that one.
    # See https://issues.redhat.com/browse/RHEL-56080
    source /usr/lib/os-release
    if [ -f /usr/libexec/ignition-sgdisk ]; then
        inst /usr/libexec/ignition-sgdisk /usr/sbin/sgdisk
    else
        inst sgdisk
    fi
```